### PR TITLE
Merge main with collaboration service branch + conflict revolve

### PR DESCRIPTION
**Pull request by whitesnowx**
Merge collaboration-branch
*  `collaboration-branch` as of commit: bab5d36f193c863cb002d21514eb60d3fd54b05d

Conflicting Files:
* `backend/matching-service/handler/socketHandler.js`
* `frontend/src/pages/matching-service/FindingMatch.js`

### CODE REVIEW

File: frontend/src/pages/matching-service/FindingMatch.js [hunk 1]
```diff
@@ -17,35 +18,31 @@ function FindingMatch() {
...
+    event.preventDefault();
+    var confirmation = window.confirm("You are exiting the matching queue, continue?");
+    if (confirmation) {
+      matchingSocket.emit("cancel_matching", { topic, difficultyLevel, email, token, username, isAny: isAnyDifficulty });
+      location.state = "";
+      navigate("/matching/select");
+    } else {
+      event.stopImmediatePropagation();
+      event.preventDefault();
+    }
```
Comment (tsulim) [inline frontend/src/pages/matching-service/FindingMatch.js:35]: Revert the changes here thank you 🙏

**Review by tsulim — COMMENTED** [review]
On overall, it looks good to me.
Minor changes required before merging. Thank you 🙏

File: frontend/src/pages/matching-service/FindingMatch.js (hunk 1 above)
Comment (whitesnowx) [reply frontend/src/pages/matching-service/FindingMatch.js]: amended

File: backend/collaboration-service/.env.sample [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Port numbers
+PORT=5003
+
+# Firebase
+PRIVATE_KEY=
+
+# Rabbit MQ
+RABBIT_PASSWORD=
```
[the remark this answers is not in the document]
Comment (whitesnowx) [reply backend/collaboration-service/.env.sample]: ammended

**Review by tsulim — APPROVED** [review]
LGTM 🔥👍
